pkg/controllers/nodeclass/status/readiness.go

```diff
@@ -33,30 +34,30 @@ type Readiness struct {
 
 func (n Readiness) Reconcile(ctx context.Context, nodeClass *v1beta1.EC2NodeClass) (reconcile.Result, error) {
 	if len(nodeClass.Status.AMIs) == 0 {
-		nodeClass.StatusConditions().SetFalse(v1beta1.ConditionTypeNodeClassReady, "NodeClassNotReady", "unable to resolve AMIs")
-		return reconcile.Result{Requeue: true}, nil
+		nodeClass.StatusConditions().SetFalse(status.ConditionReady, "NodeClassNotReady", "Failed to resolve AMIs")
```

Comment:
Can the reason in each of these match the actual reason why it's not Ready? Such as `AMIDiscoveryFailed` or `SubnetDiscoveryFailed`, etc.